compute/disks/disks_test.go

```diff
@@ -465,4 +498,184 @@ func TestComputeDisksSnippets(t *testing.T) {
+			}
+		}()
+
+		// Verify the storage pool creation
+		storagePool, err := storagePoolsClient.Get(ctx, &computepb.GetStoragePoolRequest{
+			Project:     tc.ProjectID,
+			Zone:        zone,
+			StoragePool: storagePoolName,
+		})
+		if err != nil {
+			t.Errorf("Get storage pool got err: %v", err)
+		}
+
+		if storagePool.GetName() != storagePoolName {
+			t.Errorf("Storage pool name mismatch: got %v, want %v", storagePool.GetName(), storagePoolName)
+		}
+
+		if !strings.Contains(storagePool.GetStoragePoolType(), "hyperdisk-balanced") {
+			t.Errorf("Storage pool type mismatch: got %v, want to contain %v", storagePool.GetStoragePoolType(), "hyperdisk-balanced")
+		}
+
+		if storagePool.GetCapacityProvisioningType() != capacityProvisioningType {
+			t.Errorf("Capacity provisioning type mismatch: got %v, want %v", storagePool.GetCapacityProvisioningType(), capacityProvisioningType)
+		}
+
+		if storagePool.GetPoolProvisionedCapacityGb() != provisionedCapacity {
+			t.Errorf("Provisioned capacity mismatch: got %v, want %v", storagePool.GetPoolProvisionedCapacityGb(), provisionedCapacity)
+		}
+
+		if storagePool.GetPoolProvisionedIops() != provisionedIops {
+			t.Errorf("Provisioned IOPS mismatch: got %v, want %v", storagePool.GetPoolProvisionedIops(), provisionedIops)
+		}
+
+		if storagePool.GetPoolProvisionedThroughput() != provisionedThroughput {
+			t.Errorf("Provisioned throughput mismatch: got %v, want %v", storagePool.GetPoolProvisionedThroughput(), provisionedThroughput)
+		}
+	})
+}
+func TestCreateDiskInStoragePool(t *testing.T) {
+	ctx := context.Background()
+	var r = rand.New(rand.NewSource(time.Now().UnixNano()))
+	tc := testutil.SystemTest(t)
+	zone := "europe-west4-b"
+	storagePoolName := fmt.Sprintf("test-storage-pool-%v-%v", time.Now().Format("01-02-2006"), r.Int())
+	storagePoolType := fmt.Sprintf("projects/%s/zones/%s/storagePoolTypes/hyperdisk-balanced", tc.ProjectID, zone)
+
+	storagePoolLink := fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/zones/%s/storagePools/%s", tc.ProjectID, zone, storagePoolName)
+	diskName := fmt.Sprintf("test-disk-%v-%v", time.Now().Format("01-02-2006"), r.Int())
+	diskType := fmt.Sprintf("zones/%s/diskTypes/hyperdisk-balanced", zone)
+	diskSizeGb := int64(50)
+	provisionedCapacity := int64(10240)
+	provisionedIops := int64(10000)
+	provisionedThroughput := int64(1024)
+
+	capacityProvisioningType := "ADVANCED"
+
+	// Create the storage pool
+	var buf bytes.Buffer
+	err := createHyperdiskStoragePool(&buf, tc.ProjectID, zone, storagePoolName, storagePoolType, capacityProvisioningType, provisionedCapacity, provisionedIops, provisionedThroughput)
+	if err != nil {
+		t.Fatalf("createHyperdiskStoragePool got err: %v", err)
+	}
+
+	defer func() {
+		if err := deleteStoragePool(tc.ProjectID, zone, storagePoolName); err != nil {
+			t.Fatalf("deleteStoragePool got err: %v", err)
+		}
+	}()
+
+	t.Run("CreateDiskInStoragePool", func(t *testing.T) {
```

Comment:
optional: you could refactor this to use the same storage pool from your previous test example, if you wanted to.